src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -214,6 +215,25 @@ internal partial class Binder
             { WasCompilerGenerated = wasCompilerGenerated };
         }
 
+        internal void CheckConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNodeOrToken syntax, Conversion conversion, BindingDiagnosticBag diagnostics)
```

Comment:
I am a little curious about why this check had to move, as well as how you determined where are all the appropriate places to insert a call to this method. Are there simply other places where user-defined conversions are created that we had to account for?

Reply:
> I am a little curious about why this check had to move, as well as how you determined where are all the appropriate places to insert a call to this method. Are there simply other places where user-defined conversions are created that we had to account for?

The original check was covering only BoundConversion node. However, there are a lot of other nodes that classify conversion, store Conversion instance and emulate precence of BoundConversion in lowering. That is how I located some of the places (theer was no goal to locate them all in this PR). Also, it felt appropriate to keep the checks close to the code that reports obsolete diagnostics for conversions (ReportDiagnosticsIfObsolete).